lib/wellies_web/parser/open_weather.ex

```diff
@@ -0,0 +1,47 @@
+defmodule WelliesWeb.OpenWeatherParser do
+
+  def get_current_forecast(body) do
+     body
+     |> get_weather_list()
+     |> List.first()
+  end
+
+  def temperatures_grouped_by_date(body) do
+    body
+    |> get_weather_list()
+    |> clean_date_field()
+    |> group_by_date()
+  end
+
+  defp get_weather_list(body), do: Map.get(body, "list")
+
+  defp clean_date_field(element), do: Enum.map(element, fn x -> update_date(x) end)
+
+  defp group_by_date(element), do: Enum.group_by(element, fn x -> date_field(x) end)
+
+  defp update_date(element), do: Map.update!(element, "dt_txt", fn x -> get_date_from_datefield(x) end)
+
+  defp date_field(element), do: Map.get(element, "dt_txt")
+
+  def temperature_field(element), do: element["main"]["temp"]
+
+  defp get_date_from_datefield(datefield) do
+    datefield
+    |> String.split
```

Comment:
Just for consistency with the above functions, I add the parentheses to the function calls. AFAIK, the default Mix formatter also adds the parentheses, but if you don't like them, then you can erase them from the above?